Topics/System_Design/DesignPatterns/Creational_Design_Patterns.md

```diff
@@ -0,0 +1,126 @@
+# Creational Design Patterns
+
+## Introduction
+
+As the name suggests, creational design patterns are standardized ways to deal with the creation and maintenance of objects in object-oriented programming. They are used to instantiate different classes of objects and their subclasses in the best way to achieve their intended purposes. 
+
+There are 5 major types of creational patterns that are commonly used in software development.
+
+## Types of Creational Design Patterns
+
+### Simple Factory
+
+#### Usage
+
+A complex object needs to be created multiple times.
+
+The simple factory works by delegating the instantiation of an object class to a dedicated method. This allows the object to be hidden from the user endpoint, by preventing the need to directly instantiate (and thus access) the object.
+
+#### Structure
+
+A factory class is introduced in addition to the object class. The factory has a create method, whose purpose is to generate and return an instance of the object class.
+
+<p align="center">
+<img width="690" alt="UML of Simple Factory design pattern" src="../assets/Simple Factory.png">
+</p>
+
+#### Benefits
+
+- Hides the object class creation behind a layer of abstraction. Can be used to simplify the object creation provided a few parts need to be built before instantiation, or adjustments need to be made after instantiation.
+
+### Factory Method
+
+#### Usage
+
+A singular class of objects needs to be produced multiple times, and in a dynamic way via subclasses of the object.
+
+The factory method design pattern introduces a class containing a singular create method that creates one class of objects (products). Extensions of the factory class are created as needed to override the default implementation (or abstraction) of the create method to produce different subclasses of the product.
+
+#### Structure
+
+A factory class containing one (abstract) create method that produces a product class of the required type. Implementations of the factory override the create method as required to output subclasses of the product.
+
+<p align="center">
+<img width="690" alt="UML of Factory Method design pattern" src="../assets/Factory Method.png">
+</p>
+
+#### Benefits
+
+- Allows variations of the product class to be easily created without having to define new calling behaviour in programs utilising the product and factory classes.
+
```

Comment:
Optionally, I think adding short examples for each design pattern to illustrate what scenarios they could be used in could be a good addition! 

For example: When programming functionality for a bakery, a Simple Factory design pattern can be used to create different types of pastries such as croissants, muffins, and cookies. The factory takes input from customers (e.g., pastry type) and produces the corresponding pastry without exposing the complexities of the baking process.